#!/bin/bash


fastqc ./*.fastq

echo "----------------------------------------------------------------------------------------"
echo "----------------------------------------------------------------------------------------"
echo "-----------------------------------------First FastQC is Done---------------------------"
echo "----------------------------------------------------------------------------------------"
echo "----------------------------------------------------------------------------------------"
sleep 5

for R1 in *.fastq ;do R2="${R1%R1.fastq}R2.fastq"; trimmomatic PE $R1 $R2 "${R1%.fastq}_trimmed.fq" "${R1%.fastq}_dropped.fq" "${R2%.fastq}_trimmed.fq" "${R2%.fastq}_dropped.fq" ILLUMINACLIP:./adaptor.fa:2:30:10:2:True SLIDINGWINDOW:4:15 LEADING:3 TRAILING:3 MINLEN:100; done

echo "----------------------------------------------------------------------------------------"
echo "----------------------------------------------------------------------------------------"
echo "----------------------------------------Trimmomatic is Done-----------------------------"
echo "----------------------------------------------------------------------------------------"
echo "----------------------------------------------------------------------------------------"
sleep 5

fastqc ./*_trimmed.fq

echo "----------------------------------------------------------------------------------------"
echo "----------------------------------------------------------------------------------------"
echo "----------------------------------------Second FastQC is Done---------------------------"
echo "----------------------------------------------------------------------------------------"
echo "----------------------------------------------------------------------------------------"
sleep 5

bwa index reference.fasta

echo "----------------------------------------------------------------------------------------"
echo "----------------------------------------------------------------------------------------"
echo "----------------------------------------Indexing is Done--------------------------------"
echo "----------------------------------------------------------------------------------------"
echo "----------------------------------------------------------------------------------------"
sleep 5

for R1 in *R1_trimmed.fq ; do R2="${R1%R1_trimmed.fq}R2_trimmed.fq"; bwa mem reference.fasta $R1 $R2 > "${R1%_R1_trimmed.fq}".sam; done
echo "----------------------------------------------------------------------------------------"
echo "----------------------------------------------------------------------------------------"
echo "----------------------------------------Alignment is Done-------------------------------"
echo "----------------------------------------------------------------------------------------"
echo "----------------------------------------------------------------------------------------"
sleep 5

for i in *genomics.sam ;do samtools view -Sb $i | samtools sort -o ${i%.sam}_sorted.bam; done
echo "----------------------------------------------------------------------------------------"
echo "----------------------------------------------------------------------------------------"
echo "-------------------------Conversion of SAM to BAM is Done-------------------------------"
echo "----------------------------------------------------------------------------------------"
echo "----------------------------------------------------------------------------------------"
sleep 5

for i in *sorted.bam ;do samtools markdup $i ${i%.bam}_dedup.bam; done
echo "----------------------------------------------------------------------------------------"
echo "----------------------------------------------------------------------------------------"
echo "-------------------------------Deduplication step is Done-------------------------------"
echo "----------------------------------------------------------------------------------------"
echo "----------------------------------------------------------------------------------------"
sleep 5

for i in *_dedup.bam ;do samtools index $i; done
echo "----------------------------------------------------------------------------------------"
echo "----------------------------------------------------------------------------------------"
echo "-----------------------------------------Indexing is Done-------------------------------"
echo "----------------------------------------------------------------------------------------"
echo "----------------------------------------------------------------------------------------"
sleep 5

# 4. Perform variant calling
for i in *_dedup.bam ;do bcftools mpileup -Ou -f reference.fasta $i | bcftools call -mv -Ob -o ${i%_sorted_dedup.bam}.bcf; done
for i in *.bcf ;do bcftools view $i > ${i%.bcf}.vcf; done

echo "----------------------------------------------------------------------------------------"
echo "----------------------------------------------------------------------------------------"
echo "----------------------------------Variant Calling is Done-------------------------------"
echo "----------------------------------------------------------------------------------------"
echo "----------------------------------------------------------------------------------------"
sleep 5

for i in *.vcf;do vcftools --vcf $i --minQ 30 --recode --recode-INFO-all --out ${i%.vcf}filtered_by_quality;done
for i in *.vcf;do vcftools --vcf $i --chr chr1 --recode --recode-INFO-all --out ${i%.vcf}chr1_variants;done

echo "----------------------------------------------------------------------------------------"
echo "----------------------------------------------------------------------------------------"
echo "---------------------------------Filteration step is Done-------------------------------"
echo "----------------------------------------------------------------------------------------"
echo "----------------------------------------------------------------------------------------"
sleep 5

#For IGV visulization
bgzip *.vcf   		    # Compress the VCF file using bgzip
tabix -p vcf *.vcf.gz       # Index the VCF file using tabix

echo "----------------------------------------------------------------------------------------"
echo "----------------------------------------------------------------------------------------"
echo "------------------------Files are ready to be IGV visualized----------------------------"
echo "----------------------------------------------------------------------------------------"
echo "----------------------------------------------------------------------------------------"
sleep 5
echo "----------------------------------------------------------------------------------------"
echo "-------------------------------------GOOD NIGHT CYA-------------------------------------"
echo "----------------------------------------------------------------------------------------"

#Pesudo VCF file
#echo -e "##fileformat=VCFv4.2\n##source=SimulatedVCF\n##reference=GRCh38\n#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\tFORMAT\tSample1\tSample2\tSample3\nchr1\t10177\t.\tA\tC\t50\tPASS\tDP=10;AF=0.1\tGT:DP:GQ\t0/1:10:30\t0/0:15:45\t1/1:8:20\nchr1\t10583\t.\tC\tT\t30\tPASS\tDP=12;AF=0.2\tGT:DP:GQ\t0/1:12:40\t0/1:14:35\t0/0:20:60\nchr1\t10611\t.\tG\tA\t40\tPASS\tDP=18;AF=0.3\tGT:DP:GQ\t0/0:18:50\t0/1:10:30\t1/1:15:25\nchr1\t13302\t.\tT\tG\t20\tPASS\tDP=8;AF=0.05\tGT:DP:GQ\t0/0:8:20\t1/1:5:10\t0/1:7:15\nchr2\t20185\t.\tA\tT\t60\tPASS\tDP=22;AF=0.4\tGT:DP:GQ\t1/1:22:60\t0/1:18:50\t0/0:20:70" > example1.vcf
